src/desktop/apps/authentication/__tests__/helpers.jest.js

```diff
@@ -239,28 +246,141 @@ describe("Authentication Helpers", () => {
       })
     })
   })
+
+  describe("#apiAuthWithRedirectUrl", () => {
+    afterEach(() => {
+      jest.clearAllMocks()
+    })
+
+    describe("when there isn't a current user", () => {
+      it("returns the app url, not an api url", async () => {
+        const response = {}
+        const redirectPath = new URL("/any-path", "https://app.example.com")
+
+        const actual = await apiAuthWithRedirectUrl(response, redirectPath)
+
+        expect(actual.toString()).toMatch("https://app.example.com/any-path")
+      })
+    })
+
+    describe("when there is a current user", () => {
+      describe("when we can get a trust token from the api", () => {
+        beforeEach(() => {
+          global.fetch = jest.fn(() =>
+            Promise.resolve({
+              status: 200,
+              ok: true,
+              json: () =>
+                Promise.resolve({
+                  trust_token: "a-trust-token",
+                  issued_at: "some-datetime",
+                  expires_in: "some-datetime",
+                }),
+            })
+          )
+        })
+
+        it("returns an API URL", async () => {
+          const response = { user: { accessToken: "some-access-token" } }
+          const redirectPath = new URL("/any-path", "https://app.example.com")
+
+          const actual = await apiAuthWithRedirectUrl(response, redirectPath)
+
+          expect(actual.toString()).toMatch("https://api.example.com")
+        })
+
+        it("returns with an application URL as the redirect uri", async () => {
+          const response = { user: { accessToken: "some-access-token" } }
+          const redirectPath = new URL("/any-path", "https://app.example.com")
+
+          const actual = await apiAuthWithRedirectUrl(response, redirectPath)
+
+          const expectedRedirectUri = encodeURIComponent(
+            "https://app.example.com/any-path"
+          )
+          expect(actual.toString()).toMatch(
+            `redirect_uri=${expectedRedirectUri}`
+          )
+        })
+
+        it("returns with the trust token from the api", async () => {
+          const response = { user: { accessToken: "some-access-token" } }
+          const redirectPath = new URL("/any-path", "https://app.example.com")
+
+          const actual = await apiAuthWithRedirectUrl(response, redirectPath)
+
+          const expectedRedirectUri = "https://app.example.com/any-path"
```

Comment:
#minor. unused variable.

Reply:
🦅 👁 – thanks!

Reply:
Thinking about this again and I'm wondering if any linting can help catch this. I don't have much context for Force, and maybe @damassi can provide some insights.